_search-plugins/searching-data/retrieve-specific-fields.md

```diff
@@ -0,0 +1,908 @@
+
+Consider a`products` index containing the following document:
+
+```json
+{
+  "product_id": "123",
+  "name": "Smartphone",
+  "category": "Electronics",
+  "price": 699.99,
+  "description": "A powerful smartphone with a sleek design.",
+  "reviews": [
+    {
+      "user": "john_doe",
+      "rating": 5,
+      "comment": "Great phone!",
+      "date": "2023-01-01"
+    },
+    {
+      "user": "jane_doe",
+      "rating": 4,
+      "comment": "Good value for money.",
+      "date": "2023-02-15"
+    }
+  ],
+  "supplier": {
+    "name": "TechCorp",
+    "contact_email": "support@techcorp.com",
+    "address": {
+      "street": "123 Tech St",
+      "city": "Techville",
+      "zipcode": "12345"
+    }
+  },
+  "inventory": {
+    "stock": 50,
+    "warehouse_location": "A1"
+  }
+}
+```
+
+You want to perform a search on this index but include only the `name`, `price`, `reviews`, and `supplier` fields in the response. Additionally, you want to exclude any `contact_email` fields from the `supplier` object and `comment` fields from the `reviews` object. To achieve this, run the following search:
+
+```json
+GET /products/_search
+{
+  "_source": {
+    "includes": ["name", "price", "reviews.*", "supplier.*"],
+    "excludes": ["reviews.comment", "supplier.contact_email"]
+  },
+  "query": {
+    "match": {
+      "category": "Electronics"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Expected response:
+
+```json
+{
+  "hits": {
+    "hits": [
+      {
+        "_source": {
+          "name": "Smartphone",
+          "price": 699.99,
+          "reviews": [
+            {
+              "user": "john_doe",
+              "rating": 5,
+              "date": "2023-01-01"
+            },
+            {
+              "user": "jane_doe",
+              "rating": 4,
+              "date": "2023-02-15"
+            }
+          ],
+          "supplier": {
+            "name": "TechCorp",
+            "address": {
+              "street": "123 Tech St",
+              "city": "Techville",
+              "zipcode": "12345"
+            }
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+## Using scripted fields
+
+The `script_fields` parameter allows you to include custom fields whose values are computed using scripts in your search results. This can be useful for calculating values dynamically based on the data in the document. You can also retrieve `derived fields` using a similar approach. For more information, see [Retrieving fields]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/derived/#retrieving-fields).
+
+The following example demonstrates using `script_fields`.
+
+Let's say you have an index of products, and each product document contains the fields `price` and `discount_percentage`. You want to include a custom field in the search results that shows the discounted price of each product.
```

Comment:
Again, "Let's say" and "You want to" are not our usual voice. There isn't anything "wrong" here, but we would ideally phrase this a little differently (a little more formally).